src/db_impl.cc

```diff
@@ -1230,14 +1243,29 @@ void TitanDBImpl::OnCompactionCompleted(
         continue;
       }
       if (file->file_state() == BlobFileMeta::FileState::kPendingLSM) {
-        if (delta < 0) {
-          // Cannot happen..
-          ROCKS_LOG_WARN(db_options_.info_log,
-                         "OnCompactionCompleted[%d]: New blob file %" PRIu64
-                         " live size being negative",
-                         compaction_job_info.job_id, file_number);
-          assert(false);
-          delta = 0;
+        if (delta <= 0) {
+          // Because the flushed SST is added to superversion before
```

Comment:
It seems to me the same could happen to compaction-generated blob files. Update the comment to reflect that?